analyses/rna-seq-composition/rna-seq-composition.Rmd

```diff
@@ -0,0 +1,584 @@
+
+# How many read type fractions are outside ref range? 
+# (total = 4 * n(samples))
+table(read_fracs_with_thresholds$within_limits)
+
+# How many samples in each MEND depth group were outside ref range?
+tabyl(depth_status, within_read_type_limits_label, status)
+
+threshold_flag_annotations <- read_fracs_with_thresholds %>% 
+  left_join(read_type_codes, by="read_type_name") %>%
+  mutate(threshold_code = ifelse(within_limits == "within_limits", NA, read_type_abbrev)) %>%
+  group_by(sample_id) %>%
+  summarize(flagged_thresholds = paste(na.omit(threshold_code), collapse=", "))
+
+read_fracs_with_thresholds <- read_fracs_with_thresholds %>% 
+  left_join(threshold_flag_annotations, by="sample_id")
+
+```
+
+# Prepare data for summary plot
+## Specify labels and levels
+```{r}
+
+stat_names = c("frac_unmapped_of_total", "frac_multi_mapped_of_mapped", "frac_dupe_of_mapped", "frac_non_exonic_of_non_dupe") 
+stat_label = c("frac unmapped", "frac multi-mapped \n(of mapped)", "frac duplicate \n(of mapped)", "frac non-exonic \n(of non-duplicate)")
+names(stat_label) = stat_names
+
+
+limit_status_levels <- c("below_limit", "within_limits", "above_limit")
+limit_status_colors <- c("white", "grey", "black")
+names(limit_status_colors) <- limit_status_levels
+
+MEND_depth_codes <- c("<10M_MEND_reads", "10-20 MEND_reads", ">20M_MEND_reads")
+MEND_depth_colors <- c("red", "pink", "springgreen4")
+names(MEND_depth_colors) <- MEND_depth_codes
+
+limit_and_MEND_vec <- c(limit_status_colors, MEND_depth_colors)
+
+```
+
+## Add factors and prepare text
+```{r}
+read_fracs_with_thresholds_anno <- read_fracs_with_thresholds %>%
+  mutate(read_type_name = factor(read_type_name, levels = read_type_codes$read_type_name),
+         within_limits = factor(within_limits, levels = limit_status_levels))
+
+```
+
+
+## Tabulate excluded samples
+```{r}
+
+flag_summary_stats <- read_fracs_with_thresholds_anno  %>%
+  mutate(`Fraction of sample` = gsub("\n", "", stat_label[match(read_type_name, names(stat_label))])) %>%
+  #  group_by(read_type_name) %>%
+  tabyl(`Fraction of sample`, within_limits) %>%
+  adorn_totals(where = c("col")) %>%
+  adorn_percentages(denominator = "row") %>% 
+  adorn_pct_formatting(digits = 0) %>%
+  adorn_ns() %>%
+  mutate(`Fraction of sample` = factor(`Fraction of sample`, 
+                                       levels = gsub("\n", "", stat_label))) %>%
+  arrange(`Fraction of sample`)
+```
+
+
+## Histogram
+```{r}
+
+plot_title <- paste0(sum(threshold_flag_annotations$flagged_thresholds!=""), " samples are outside reference ranges")
+
+p20 <-  ggplot(read_fracs_with_thresholds_anno) + 
+  geom_histogram(aes(x = sample_value, fill = read_type_name)) + 
+  geom_vline(data=all_read_type_fraction_stats,
+             aes(xintercept = value, linetype = stat, color = read_type_name)) +
+    scale_fill_viridis_d() +
+  scale_color_viridis_d() +
+  facet_wrap(~read_type_name, nrow=1, labeller = labeller(
+    read_type_name=stat_label) )
```

Comment:
I think having a free X scale may make these easier to read, but it may be misleading. I'll leave it to you decide which you prefer.
```suggestion
  facet_wrap(~read_type_name, 
             nrow=1,
             scales="free_x",
             labeller = labeller(read_type_name=stat_label) )
```